src/analyz/ty_json.rs

```diff
@@ -412,35 +419,35 @@ impl<'tcx> ToJson<'tcx> for ty::Ty<'tcx> {
             &ty::TyKind::Closure(defid, ref substs) => {
                 json!({
                     "kind": "Closure",
-                    "upvar_tys": substs.as_closure().upvar_tys(defid, mir.state.tcx)
+                    "upvar_tys": substs.as_closure().upvar_tys()
                         .collect::<Vec<_>>().to_json(mir),
                     // crucible-mir uses the same representation for closures as it does for
                     // tuples, so no additional information is needed.
                 })
             }
-            &ty::TyKind::Dynamic(ref preds, _region) => {
-                let ti = TraitInst::from_dynamic_predicates(mir.state.tcx, *preds);
+            &ty::TyKind::Dynamic(preds, _region, _dynkind) => {
+                let ti = TraitInst::from_dynamic_predicates(mir.state.tcx, preds);
```

Comment:
Maybe we should require the last field here to be `DynKind::Dyn` (and have a separate case similar to `TyKind::Error` for `DynKind::DynStar`).  It's better for code using `dyn*` to fail explicitly rather than trigger some weird interaction between this code and the `TraitInst` code above that only handles the `DynKind::Dyn` case.

Reply:
Sorry, I think I'm missing something. What `TraintInst` code are you referring to?

Reply:
Ah, perhaps you are referring to this code?

https://github.com/GaloisInc/mir-json/blob/be4359a4e4361dd2c403e30b51fccc421f8304e4/src/analyz/to_json.rs#L155-L165

I will see what happens if I explicitly rule out `DynKind::DynStar` (hopefully no test cases start failing).

Reply:
Yes, that's the code I meant

Reply:
I've split this into separate `Dyn` and `DynStar` cases.